components/builder-graph/src/acyclic_package_graph.rs

```diff
@@ -234,10 +247,163 @@ impl PackageGraphTrait for AcyclicPackageGraph {
                 connected_comp: connected_components(&self.graph),
                 is_cyclic:      is_cyclic_directed(&self.graph), }
     }
+
+    // compute_build for AcyclicGraph
+    // Process
+    // 1) Take kernel of packages, and recursively expand it over reverse build/runtime deps
+    //    creating a new sub-graph.
+    // 2) Filter unbuildable nodes in subgraph, flag them as directly
+    //    (no plan connection, autobuilds off) unbuildable
+    // 3) Use the unbuildable packages as kernel, expand it over the subgraph to
+    //    create a list of all unbuildable packages.
+    // 4) Iterate over all unbuildable packages.
+    //    Flag new entries as indirectly unbuildable.
+    //    Remove that package from the sub-graph
+    // 5) Walk the remaining sub-graph, collecting dependencies that are external to
+    //    the sub-graph.
+    // 6) Create a PackageBuildManifestGraph using the information generated in 1-5.
+    fn compute_build(&self,
+                     touched: &[PackageIdentIntern],
+                     unbuildable: &dyn Unbuildable)
+                     -> Result<PackageBuildManifest> {
+        let mut rebuild_graph: DiGraphMap<PackageIdentIntern, EdgeType> = DiGraphMap::new();
+        let mut unbuildable_map: HashMap<PackageIdentIntern, UnbuildableReason> = HashMap::new();
+        let mut all_external_dependencies: HashSet<PackageIdentIntern> = HashSet::new();
+
+        let mut worklist: VecDeque<PackageIdentIntern> =
```

Comment:
Is there a particular reason for using `VecDeque` over a plain old `Vec`? It looks like you only ever interact with one end of the data structure.